src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -843,6 +850,25 @@ private static Task ProcessUnknownFrameAsync()
             {
                 throw new Http3StreamErrorException(CoreStrings.FormatHttp3DatagramStatusMismatch(_context.ClientPeerSettings.H3Datagram == 1, _context.ServerPeerSettings.H3Datagram == 1), Http3ErrorCode.SettingsError);
             }
+
+            if (string.Equals(HttpRequestHeaders.HeaderProtocol, WebTransportSession.WebTransportProtocolValue, StringComparison.Ordinal))
+            {
+                // indicate as webtransport request
+                _currentIHttpWebTransportFeature!.IsWebTransportRequest = true;
+
+                // TODO do as we go along and store as list in WebTransportSession or is that a memory issue?
+                // Also, should this be on a per webtransport session basis? If so, maybe I need a queue?
+                var supportedVersions = HttpRequestHeaders.Where((header) =>
+                {
+                    return header.Key.StartsWith(WebTransportSession.VersionHeaderPrefix, StringComparison.Ordinal)
+                            && string.Equals(header.Value, "1", StringComparison.Ordinal);
+                }).Select((header) => header.Key);
```

Comment:
I don't know if it's likely for us to support multiple versions at a time. The version check would be much simpler if you were explicitly checking for one or two versions. You wouldn't have to enumerate all the headers and do StartsWith.

Reply:
Removing this layer of complexity as it probably isn't worth it.